src/jouwid.mjs

```diff
@@ -151,35 +155,43 @@ export async function logout(options) {
  * Get a resource that is protected, so you must be logged
  * in to access it, and have an access grant.
  */
-export async function getProtectedResource(options) {
+export async function getProtectedResource(options)
+{
 	const defaultOptions = {
 	}
+
 	const validOptions = {
 		resourcePath: Required(String)
 	}
+
+    assert(options, validOptions)
+
 	options = Object.assign({}, defaultOptions, options)
-	assert(options, validOptions)
 
 	if (remoteClient && user?.id) {
-
 		const pod = await remoteClient
 		.getUsersServices()
 		.getPodInstance(user.id, MASTER_POD_ALIAS)
 
         const resource = await pod
 		.getFile(options.resourcePath)
 
-		//TODO: test that pod can convert to jsonld
-		// and make sure to always request jsonld
-		return await resource.content.text()
-	}
+        if (resource.metadata.contentType.match(/^application\/(.*\+)?json/)) {
+            return resource.content.json()
+        } else if (resource.metadata.contentType.match(/^text\//)) {
+            return resource.content.text()
+        } else {
+            return resource.content // Blob
+        }
+    }
     return false
 }
 
 /**
  * Token storage helper
  */
-const tokenStorage = function(type) {
+const tokenStorage = function(type)
```

Comment:
Moeten we deze hier ook naar een utils folder verplaatsen?

Reply:
Hm, in de war met de datakluis/components repo, hier is nog geen reden om de tokenstorage naar een eigen file te zetten, aangezien er nog geen andere javascript modules zijn die het nodig hebben... policy bij ons is om dingen op te splitsen op het moment dat het nodig is, niet eerder, zou ik hier ook willen handhaven.